lib/modules/betteReddit.js

```diff
@@ -595,71 +593,52 @@ addModule('betteReddit', function(module, moduleID) {
-
-		// the subreddit manager code changes the document's structure
 		var sm = modules['subredditManager'].isEnabled();
-
 		var header = document.getElementById('header');
-		if (header === null) {
-			return; // reddit is under heavy load
-		}
-
-		// add a dummy <div> to the document for spacing
-		var spacer = document.createElement('div');
-		spacer.id = 'RESPinnedHeaderSpacer';
 
-		// without the next line, the subreddit manager would make the subreddit bar three lines tall and very narrow
-		RESUtils.addCSS('#sr-header-area {left: 0; right: 0;}');
-		spacer.style.height = $('#header').outerHeight() + 'px';
+		if (header === null) return; // Reddit is having issues.
 
-		// insert the spacer
-		document.body.insertBefore(spacer, header.nextSibling);
+		var headerHeight = $('#header').outerHeight(true);
+		var spacer = RESUtils.createElement('div', 'RESPinnedHeaderSpacer');
 
-		// make the header fixed
-		RESUtils.addCSS('#header, #RESAccountSwitcherDropdown {position:fixed;}');
-		// RESUtils.addCSS('#header {left: 0; right: 0; box-shadow: 0 2px 2px #AAA;}');
-		RESUtils.addCSS('#header {left: 0; right: 0; }');
-		var headerHeight = $('#header').height() + 15;
-		RESUtils.addCSS('#RESNotifications { top: ' + headerHeight + 'px } ');
+		// Add fixed positioning to all relevant elements.
+		RESUtils.addCSS('#header, #RESAccountSwitcherDropdown { position: fixed; }');
 		pinCommonElements(sm);
 
-		// TODO Needs testing
-		// Sometimes this gets executed before the subreddit logo has finished loading. When that
-		// happens, the spacer gets created too short, so when the SR logo finally loads, the header
-		// grows and overlaps the top of the page, potentially obscuring the first link. This checks
-		// to see if the image is finished loading. If it is, then the spacer's height is set. Otherwise,
-		// it pauses, then loops.
-		// added a check that this element exists, so it doesn't error out RES.
-		if (document.getElementById('header-img') && (!document.getElementById('header-img').complete)) {
-			setTimeout(function() {
-				if (document.getElementById('header-img').complete) {
-					// null parameter is necessary for FF3.6 compatibility.
-					document.getElementById('RESPinnedHeaderSpacer').style.height = window.getComputedStyle(document.getElementById('header'), null).height;
-				} else {
-					setTimeout(arguments.callee, 10);
-				}
-			}, 10);
-		}
+		// Ensure the header remains in its original position at the top of the page.
+		// We can't change the z-index because themes often rely on it being a certain value.
+		RESUtils.addCSS('#header { top: 0; left: 0; right: 0; bottom: auto; }');
+
+		// Spacer height should be equal to header (including padding, margins, etc).
+		$(spacer).height(headerHeight);
+		document.body.insertBefore(spacer, header.nextSibling);
+
+		// Hide sidebar checkbox.
+		setTimeout(function () {
+			modules['styleTweaks'].setSRStyleToggleVisibility(false, 'pinHeader');
```

Comment:
Probably want to avoid doing this, if possible (unless it's shown somewhere else). Does it overlap with the header or something?